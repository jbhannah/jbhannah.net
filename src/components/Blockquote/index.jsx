import React from "react"
import { MOBILE_MEDIA_QUERY } from "typography-breakpoint-constants"
import { linkColor, mq } from "../../utils/styles"
import { rhythm } from "../../utils/typography"

const borderLeft = i => `${rhythm(i / 16)} solid ${linkColor}`

const baseBlockquote = {
  borderLeft: borderLeft(3),
  marginLeft: rhythm(-3 / 4),
  paddingLeft: rhythm(9 / 16),
}

const blockquoteCSS = {
  ...baseBlockquote,
  [MOBILE_MEDIA_QUERY]: { ...baseBlockquote },
  [mq.sm]: {
    borderLeft: borderLeft(6),
    marginLeft: 0,
    paddingLeft: rhythm(10 / 16),
  },
}

const Blockquote = props => <blockquote css={blockquoteCSS} {...props} />

export default Blockquote
