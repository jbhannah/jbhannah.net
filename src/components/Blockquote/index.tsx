import * as React from "react"
import { MOBILE_MEDIA_QUERY } from "typography-breakpoint-constants"
import { linkColor, mq } from "../../utils/styles"
import { rhythm } from "../../utils/typography"

const borderLeft = (i: number) => `${rhythm(i / 16)} solid ${linkColor}`

const baseBlockquote = {
  borderLeft: borderLeft(3),
  marginLeft: rhythm(-3 / 4),
  paddingLeft: rhythm(9 / 16),
}

const blockquoteCSS = {
  ...baseBlockquote,
  [MOBILE_MEDIA_QUERY]: { ...baseBlockquote },
  [mq.sm]: {
    borderLeft: borderLeft(6),
    marginLeft: 0,
    paddingLeft: rhythm(10 / 16),
  },
}

type BlockquoteProps = JSX.IntrinsicAttributes &
  React.ClassAttributes<HTMLElement> &
  React.BlockquoteHTMLAttributes<HTMLElement>

export const Blockquote: React.FunctionComponent<BlockquoteProps> = (props) => (
  <blockquote css={blockquoteCSS} {...props} />
)
