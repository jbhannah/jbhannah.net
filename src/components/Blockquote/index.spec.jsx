import { shallow } from "enzyme"
import React from "react"
import Blockquote from "."

const TestBlockquote = props => shallow(<Blockquote {...props} />)

describe("Blockquote", () => {
  const tree = TestBlockquote()

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot()
  })
})
