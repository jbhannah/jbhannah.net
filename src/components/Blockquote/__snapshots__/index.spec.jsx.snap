// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote renders correctly 1`] = `
<blockquote
  css={
    Object {
      "@media only screen and (max-width:480px)": Object {
        "borderLeft": "0.29625rem solid #0f7b3b",
        "marginLeft": "-1.185rem",
        "paddingLeft": "0.88875rem",
      },
      "@media only screen and (min-width: 28rem)": Object {
        "borderLeft": "0.5925rem solid #0f7b3b",
        "marginLeft": 0,
        "paddingLeft": "0.9875rem",
      },
      "borderLeft": "0.29625rem solid #0f7b3b",
      "marginLeft": "-1.185rem",
      "paddingLeft": "0.88875rem",
    }
  }
/>
`;
